jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt

```diff
@@ -84,35 +85,16 @@ class SamRunningState(
         val lambdaSettings = samState.settings
         val lambdaBuilder = LambdaBuilder.getInstance(lambdaSettings.runtimeGroup)
 
-        val buildLambdaRequest = when (lambdaSettings) {
-            is TemplateRunSettings ->
-                buildLambdaFromTemplate(
-                    lambdaSettings.templateFile,
-                    lambdaSettings.logicalId,
-                    lambdaSettings.samOptions
-                )
-            is ImageTemplateRunSettings ->
-                buildLambdaFromTemplate(
-                    lambdaSettings.templateFile,
-                    lambdaSettings.logicalId,
-                    lambdaSettings.samOptions
-                )
-            is HandlerRunSettings ->
-                buildLambdaFromHandler(
-                    lambdaBuilder,
-                    environment.project,
-                    lambdaSettings
-                )
-        }
+        val buildLambdaRequest = buildBuildLambdaRequest(environment.project, lambdaSettings)
 
         samState.pathMappings = createPathMappings(lambdaBuilder, lambdaSettings, buildLambdaRequest)
         val buildWorkflow = buildWorkflow(environment, settings, samState, buildLambdaRequest, buildView)
 
         return DefaultExecutionResult(buildView, buildWorkflow)
     }
 
-    private fun createPathMappings(lambdaBuilder: LambdaBuilder, settings: LocalLambdaRunSettings, buildRequest: BuildRequest): List<PathMapping> {
-        val defaultPathMappings = lambdaBuilder.defaultPathMappings(buildRequest.template, buildRequest.logicalId, buildRequest.buildDir)
+    private fun createPathMappings(lambdaBuilder: LambdaBuilder, settings: LocalLambdaRunSettings, buildRequest: BuildLambdaRequest): List<PathMapping> {
+        val defaultPathMappings = lambdaBuilder.defaultPathMappings(buildRequest.templatePath, buildRequest.logicalId ?: dummyLogicalId, buildRequest.buildDir)
```

Comment:
When is buildRequest.logicalId null?

Reply:
it's nullable through code in LambdaWorkflows.kt

Reply:
yaaaa I saw that later :(

Is it worth making a not null version that extends the BuildLambda? BuildLambdaWithId?